packages/gensx/src/checkpoint.ts

```diff
@@ -80,6 +83,25 @@ export class CheckpointManager implements CheckpointWriter {
       checkpointsEnv === "true" ||
       checkpointsEnv === "1" ||
       checkpointsEnv === "yes";
+
+    if (!this.checkpointsEnabled) {
+      return;
+    }
+
+    if (!process.env.GENSX_ORG) {
+      throw new Error(
+        "GENSX_ORG is not set, must be set to record checkpoints.",
+      );
+    }
+
+    this.org = process.env.GENSX_ORG;
+
+    if (!process.env.GENSX_API_KEY) {
+      throw new Error(
+        "GENSX_API_KEY is not set, must be set to record checkpoints.",
+      );
+    }
+    this.apiKey = process.env.GENSX_API_KEY;
```

Comment:
I think we should adjust the way these flags compose now that we have an API key: 

1. presence of API key determines whether or not checkpoints are on
2. GENSX_CHECKPOINTS defaults to true, but setting it to false turns the feature off even if API is set. 

Feels weird to have to set: `GENSX_API_KEY=.. GENSX_CHECKPOINTS=true node ...`

Reply:
I like that, let me make that change.